be/src/exprs/binary_predicate.cpp

```diff
@@ -117,8 +117,18 @@ class ArrayPredicate final : public Predicate {
     StatusOr<ColumnPtr> evaluate_checked(ExprContext* context, Chunk* ptr) override {
         ASSIGN_OR_RETURN(auto l, _children[0]->evaluate_checked(context, ptr));
         ASSIGN_OR_RETURN(auto r, _children[1]->evaluate_checked(context, ptr));
-        auto lhs_arr = std::static_pointer_cast<ArrayColumn>(l);
-        auto rhs_arr = std::static_pointer_cast<ArrayColumn>(r);
+
+        if (l->only_null() || r->only_null()) {
```

Comment:
for null-safe eq, here must return all trues.

Reply:
added another predicate to deal with null safe eq